src/proxy/inbound.rs

```diff
@@ -250,8 +283,10 @@ impl Inbound {
         ri.result_tracker.record(res);
     }
 
+    // build_inbound_request builds up the context for an inbound request.
     async fn build_inbound_request(
```

Comment:
@howardjohn Any suggestions for testing `build_inbound_request` or it best to break out the parts we want to test into separate methods? Is testing for methods like these typically captured in the istio tests?

Reply:
The hard part of testing this is likely making a `H2Request`? Since that would pretty much need to come from a real connection which makes it more e2e test than a small unit test.

We only actually need properties of the request (uri/method/headers) though - in theory we could make the function take in `Parts` (not currently exposed) only which would make it pretty simple to have just simple unit tests. Might be worth doing so...

Reply:
Updated build_inbound_request to accept  `Parts` instead of `H2Request`. To simplify testing even further, I could define a trait and implement it for `Parts` and a mock `Parts` since I can't create a Parts instance directly (I'm currently creating a `Request` and accessing the `Parts` in the test. Both options are relatively simple/minimal code

Reply:
Never mind, added a `RequestParts` trait